core/services/relay/evm/chain_reader.go

```diff
@@ -298,6 +298,62 @@ func (cr *chainReader) QueryKey(
 	return sequenceOfValues, nil
 }
 
+func (cr *chainReader) QueryKeys(ctx context.Context, keyQueries []commontypes.KeyQuery,
```

Comment:
what is keyQuery

Reply:
its this, naming not fixed, just interested in if the general approach makes sense:
`type KeyQuery struct {
	query.KeyFilter
	Contract         BoundContract
	SequenceDataType any
}`